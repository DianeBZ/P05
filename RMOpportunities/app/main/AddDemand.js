import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TextInput,
  TouchableHighlight,
  ScrollView
} from 'react-native';

import Translation from './Translation';
import DemandConnection from './AddDemandConnection';

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;

var Step1 = true;
var Step2 = false;

var star = "*";

class MenuAddDemand extends Component
{
    constructor(props)
	{
        super(props);
        this.state = 
		{
            numIdentification : '',
            nom: '',
            numReach: '',
            categorie: '',
            description: '',
            qtteSouhaitee: '',
        };
    }

	render(){
        return(
        <ScrollView>
            {(()=>{
                if (Step1){
                    return(
                    <View>
                        <View style={{flexDirection:'row', justifyContent:'flex-start', width:width}}>
                            <View style={{borderWidth:1, borderColor:'#A4D04A', height:height*0.05}}>
                                <Text style={{fontSize:22, marginLeft:15, marginRight:15, color:"#000000"}}>{Translation[lang].etape1}</Text>
                            </View>
                            <View style={{borderWidth:1, borderColor:'#A4D04A', height:height*0.05, backgroundColor:"#F2F2F2"}}>
                                <Text style={{fontSize:22, marginLeft:15, marginRight:15, color:"#000000"}}>2</Text>
                            </View>
                            <View style={{flex:1, backgroundColor:'#F2F2F2'}}/>
                        </View>
                        <View style={{alignItems:'center'}}>
                            <TextInput style={styles.textToFill} underlineColorAndroid={'transparent'} placeholder={Translation[lang].num_id + star} onChangeText={(numID) => this.setState({numID})} value={this.state.numID}/>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={[styles.textToFill2, {width:width*0.4}]} underlineColorAndroid={'transparent'} placeholder={Translation[lang].nom_prod + star}  onChangeText={(nom) => this.setState({nom})} value={this.state.nom}/>
                                <TextInput style={styles.textToFill2} underlineColorAndroid={'transparent'} placeholder={Translation[lang].num_reach} onChangeText={(numReach) => this.setState({numReach})} value={this.state.numReach}/>
                            </View>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={[styles.textToFill2, {width:width*0.4}]} underlineColorAndroid={'transparent'} placeholder={Translation[lang].nom_douane} onChangeText={(nomenc) => this.setState({nomenc})} value={this.state.nomenc}/>
                                <TextInput style={styles.textToFill2} underlineColorAndroid={'transparent'} placeholder={Translation[lang].categorie + star} onChangeText={(categorie) => this.setState({categorie})} value={this.state.categorie}/>
                            </View>
                        </View>
                        <View style={{alignItems:'center'}}>
                            <TouchableHighlight onPress={this._changeStep} style={[styles.button, {width: width *0.5, backgroundColor:'#A4D04A'}]} underlayColor="rgb(138,183,46)">
                                <Text style={styles.buttonText}>{Translation[lang].suivant}</Text>
                            </TouchableHighlight>
                        </View>
                    </View>
                    );
                }else if (Step2){
                    return(
                    <View>
                        <View style={{flexDirection:'row', justifyContent:'flex-start', width:width}}>
                            <View style={{borderWidth:1, borderColor:'#A4D04A', height:height*0.05, backgroundColor:"#F2F2F2"}}>
                                <Text style={{fontSize:22, marginLeft:15, marginRight:15, color:"#000000"}}>1</Text>
                            </View>
                            <View style={{borderWidth:1, borderColor:'#A4D04A', height:height*0.05}}>
                                <Text style={{fontSize:22, marginLeft:15, marginRight:15, color:"#000000"}}>{Translation[lang].etape2}</Text>
                            </View>
                            <View style={{flex:1, backgroundColor:'#F2F2F2'}}/>
                        </View>
						<Text style={[styles.sousTitre, {marginLeft:12}]}>
							{Translation[lang].description + star}
						</Text>
						<View style={{alignItems:'center'}}>
                            <TextInput style={styles.textToFill3} underlineColorAndroid={'transparent'} placeholder={Translation[lang].det_desc} onChangeText={(description) => this.setState({description})} value={this.state.description}/>
                        <Text numberOfLines={3}>
							{this.state.description}
						</Text>
						</View>
						
						<Text style={[styles.sousTitre, {marginLeft:12}]}>
							{Translation[lang].qtte_souh + star}
						</Text>
						<View style={{flexDirection:'row'}}>
                                <TextInput style={[styles.textToFill2, {width:width*0.4}]} underlineColorAndroid={'transparent'} placeholder={Translation[lang].qtte} onChangeText={(qtteSouhaitee) => this.setState({qtteSouhaitee})} value={this.state.qtteSouhaitee}/>                                
                        </View>
                       
						<View style={{alignItems:'center'}}>
						<Text numberOfLines={4}>
                            <Text style={{color:'#000000'}}>{Translation[lang].texte_cgu1} </Text>
							<Text selectable={true} style={{color:'#A4D04A'}}>{Translation[lang].texte_cgu2}</Text>
							<Text style={{color:'#000000'}}>{Translation[lang].texte_cgu3}</Text>
						</Text>
                        </View>
						
						<View style={{flexDirection:'row', justifyContent:'center'}}>
                            <TouchableHighlight onPress={this._changeStep} style={[styles.button, {width: width *0.27, backgroundColor:'white', borderWidth:1, borderColor:'#A4D04A'}]}>
                                <Text style={{color:'#A4D04A', fontSize:20, textAlign:'center'}}>{Translation[lang].precedent}</Text>
                            </TouchableHighlight>
                            <TouchableHighlight onPress={this._onPressSuivant2} style={[styles.button, {width: width *0.27, backgroundColor:'#A4D04A', marginLeft:10}]} underlayColor="rgb(138,183,46)">
                                <Text style={styles.buttonText}>{Translation[lang].terminer}</Text>
                            </TouchableHighlight>
							<TouchableHighlight onPress={this._onPressAnnuler} style={[styles.button, {width: width *0.27, backgroundColor:'red', marginLeft:10}]} underlayColor="rgb(138,183,46)">
                                <Text style={styles.buttonText}>{Translation[lang].annuler}</Text>
                            </TouchableHighlight>
						</View>
                        
                    </View>
                    );
                }
            })()}
        </ScrollView>
        );
    }
	
	_changeStep = () =>
	{
        Step1 = !Step1;
        Step2 = !Step2;
        this.forceUpdate();
    };
}
	
export default class AddDemand extends Component{
    
    componentWillUpdate(){
        if (connection===0){
            const {navigator} = this.props;
            var i = 0;
            while(navigator.getCurrentRoutes()[i].name != 'AddDemand'){
                i=i+1;
            }
            if (navigator){
                navigator.replaceAtIndex({
                    name: 'DemandConnection',
                    component: DemandConnection,
                }, i)
            }
        }
    }
    
    render(){
        return(
        <ScrollView>
            <View style={{height: height * 0.08}}/>
            <View style={styles.window}>
                <Text style={styles.titre}>
                    {Translation[lang].aj_demande}
                </Text>
                <View style={styles.mainWindow}>
                    <MenuAddDemand/>
                </View>
            </View>
        </ScrollView>
        );
    }
}
const styles = StyleSheet.create({
   window:{
       width: width,
	   //height: height*0.93,
       backgroundColor: '#F2F2F2',
       alignItems: 'center',
   },
   mainWindow:{
       width: width,
       height: height,
       backgroundColor: "#FFFFFF",
       alignItems: 'center',
   },
   titre:{
       fontSize: 32,
       textAlign: 'center',
       color: '#000000',
       marginTop: 15,
       marginBottom: 10
   },
   sousTitre:{
       fontSize: 20,
       textAlign: 'left',
       color: '#000000',
       marginTop: 15,
       marginBottom: 10
   },
   button:{
        height: width*0.09,
        marginTop: 20,
        justifyContent: 'center',
        borderRadius:5,
    },
    buttonText:{
        color: "white",
        textAlign: "center",
        fontSize: 20,
    },
    textToFill:{
       width: width*0.85,
       height: width*0.1,
       borderWidth: 1,
       borderColor: "#000000",
       marginTop: 15,
       color: 'grey',
   },
   textToFill2:{
       width: width*0.35,
       height: width*0.1,
       borderWidth: 1,
       borderColor: "#000000",
       marginTop: 15,
       marginRight: width*0.05,
	   marginLeft: width*0.05,
       color: 'grey',
   },
      textToFill3:{
       width: width*0.9,
       height: height*0.2,
       borderWidth: 1,
       borderColor: "#000000",
       marginTop: 15,
       marginRight: width*0.05,
	   marginLeft: width*0.05,
       color: 'grey',
   },
});
	