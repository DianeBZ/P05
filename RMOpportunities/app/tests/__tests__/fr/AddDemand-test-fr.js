// __tests__/AddDemand-test-fr.js
import 'react-native';
import React from 'react';

global.lang = 'fr';
import AddDemand from '../../../main/AddDemand';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <AddDemand />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
