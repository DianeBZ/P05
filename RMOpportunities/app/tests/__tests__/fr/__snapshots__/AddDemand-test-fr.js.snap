// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "height": 106.72,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F2F2F2",
          "width": 750,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000000",
            "fontSize": 32,
            "marginBottom": 10,
            "marginTop": 15,
            "textAlign": "center",
          }
        }
      >
        Ajouter une demande
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "width": 750,
          }
        }
      >
        <RCTScrollView>
          <View>
            <View>
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "width": 750,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "borderColor": "#A4D04A",
                      "borderWidth": 1,
                      "height": 80.03999999999999,
                      "justifyContent": "center",
                      "width": 675,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#000000",
                        "fontSize": 22,
                        "marginLeft": 15,
                        "marginRight": 15,
                      }
                    }
                  >
                    Etape 1: Informations produit
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#F2F2F2",
                      "borderColor": "#A4D04A",
                      "borderWidth": 1,
                      "height": 80.03999999999999,
                      "justifyContent": "center",
                      "width": 75,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    onPress={[Function]}
                    style={
                      Object {
                        "color": "#000000",
                        "fontSize": 22,
                      }
                    }
                  >
                    2
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "backgroundColor": "#F2F2F2",
                      "flex": 1,
                    }
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Numéro d'indentification (CAS ou EINECS)*"
                  style={
                    Array [
                      Object {
                        "borderColor": "#000000",
                        "borderRadius": 3,
                        "borderWidth": 1,
                        "color": "grey",
                        "height": 106.72,
                        "marginTop": 15,
                        "width": 600,
                      },
                      Object {
                        "marginLeft": 75,
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "marginLeft": 75,
                    }
                  }
                >
                  <TextInput
                    onChangeText={[Function]}
                    placeholder="Nom du produit*"
                    style={
                      Array [
                        Object {
                          "borderColor": "#000000",
                          "borderRadius": 3,
                          "borderWidth": 1,
                          "color": "grey",
                          "height": 106.72,
                          "marginTop": 15,
                          "width": 600,
                        },
                        Object {
                          "width": 300,
                        },
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <TextInput
                    onChangeText={[Function]}
                    placeholder="Numéro REACH"
                    style={
                      Array [
                        Object {
                          "borderColor": "#000000",
                          "borderRadius": 3,
                          "borderWidth": 1,
                          "color": "grey",
                          "height": 106.72,
                          "marginTop": 15,
                          "width": 600,
                        },
                        Object {
                          "marginLeft": 37.5,
                          "width": 262.5,
                        },
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View>
                  <TextInput
                    onChangeText={[Function]}
                    placeholder="Nomenclature douanière"
                    style={
                      Array [
                        Object {
                          "borderColor": "#000000",
                          "borderRadius": 3,
                          "borderWidth": 1,
                          "color": "grey",
                          "height": 106.72,
                          "marginTop": 15,
                          "width": 600,
                        },
                        Object {
                          "marginLeft": 75,
                          "width": 375,
                        },
                      ]
                    }
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#000000",
                          "borderRadius": 3,
                          "borderWidth": 1,
                          "height": 106.72,
                          "marginTop": 15,
                          "width": 600,
                        },
                        Object {
                          "marginLeft": 75,
                          "width": 600,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "color": "grey",
                          "marginTop": 0,
                        }
                      }
                    >
                      <RCTPicker
                        items={
                          Array [
                            Object {
                              "label": "Choisissez une catégorie",
                              "textColor": undefined,
                              "value": "choose",
                            },
                            Object {
                              "label": "Produits pour la cosmétique, les arômes et parfums",
                              "textColor": undefined,
                              "value": "Produits pour la cosmétique, les arômes et parfums",
                            },
                            Object {
                              "label": "Solvants",
                              "textColor": undefined,
                              "value": "Solvants",
                            },
                            Object {
                              "label": "Solvants regeneres",
                              "textColor": undefined,
                              "value": "Solvants regeneres",
                            },
                            Object {
                              "label": "Reactifs",
                              "textColor": undefined,
                              "value": "Reactifs",
                            },
                            Object {
                              "label": "Catalyseurs",
                              "textColor": undefined,
                              "value": "Catalyseurs",
                            },
                            Object {
                              "label": "Specialites chimiques",
                              "textColor": undefined,
                              "value": "Specialites chimiques",
                            },
                            Object {
                              "label": "Intermediaires",
                              "textColor": undefined,
                              "value": "Intermediaires",
                            },
                            Object {
                              "label": "Intermediaires simples",
                              "textColor": undefined,
                              "value": "Intermediaires simples",
                            },
                            Object {
                              "label": "Acides",
                              "textColor": undefined,
                              "value": "Acides",
                            },
                            Object {
                              "label": "Bases",
                              "textColor": undefined,
                              "value": "Bases",
                            },
                            Object {
                              "label": "Acides amines",
                              "textColor": undefined,
                              "value": "Acides amines",
                            },
                            Object {
                              "label": "Vitamines",
                              "textColor": undefined,
                              "value": "Vitamines",
                            },
                            Object {
                              "label": "Autres",
                              "textColor": undefined,
                              "value": "Autres",
                            },
                          ]
                        }
                        onChange={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        selectedIndex={0}
                        style={
                          Array [
                            Object {
                              "height": 216,
                            },
                            undefined,
                          ]
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "marginBottom": 400.2,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Array [
                        Object {
                          "borderRadius": 5,
                          "height": 66.7,
                          "justifyContent": "center",
                          "marginTop": 20,
                        },
                        Object {
                          "backgroundColor": "#A4D04A",
                          "width": 375,
                        },
                      ],
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 20,
                        "textAlign": "center",
                      }
                    }
                  >
                    Suivant
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
